### DISCUSSION TITLE
How to exclude sensitive fields when using include

### DISCUSSION THREAD

**Discussion by jeepers3327** [prisma/prisma#8279] [Q&A] [answered]
Hi,

I'm trying to find a way to exclude a sensitive field (password) when fetching a model through `include`. I'm aware of these limitation and describe here: [https://github.com/prisma/prisma/issues/7380](https://github.com/prisma/prisma/issues/7380).

Is there any workaround with current limitation? I'm thinking of using a middle ware like below.
```ts
    this.$use(async (params, next) => {
      const data = await next(params);
      const omittedData = _.omit(data, ['author.password', 'password']);

      return omittedData;
    });
```
but it returns a different result. Instead of return a `post` object, it only returns a `user/author` object with password removed.

**Comment by ryands17** [comment, marked as answer]
@jeepers3327 👋 

It should be possible by [specifying the model](https://www.prisma.io/docs/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases) and then using `_omit` in that condition.

**Reply by jeepers3327** [reply]
@ryands17 
I've read about that and it was working fine, but when dealing with queries that have `includes` , it becomes complex and my authentication fails since the password was no longer present upon executing the service, so I decided to just explicitly defined the fields to be returned in the `include` and everything works as expected.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
